### PR TITLE
[core] Fix pidfile bug for 5.2.2

### PR DESCRIPTION
**Pull request by degemer**
And thus add the ability to be restarted properly after a SIGKILL (since
the pid in the pidfile is now tested)

### CODE REVIEW

File: daemon.py [hunk 1]
```diff
@@ -158,10 +158,23 @@ def start(self, foreground=False):
...
-            sys.stderr.write(message % self.pidfile)
-            sys.exit(1)
+            # Check if the pid in the pidfile corresponds to a running process
+            try:
+                os.kill(pid, 0)
+            except OSError, e:
+                if e.errno != errno.EPERM:
+                    log.warn('pidfile contains the pid of a not running process.'\
+                              ' Starting normally')
+                    exit_code = None
```
Comment (remh) [inline daemon.py]: Shouldn't it be 0 ?

File: daemon.py [hunk 2]
```diff
@@ -158,10 +158,23 @@ def start(self, foreground=False):
         pid = self.pid()
 
         if pid:
-            message = "pidfile %s already exists. Is it already running?\n"
-            log.error(message % self.pidfile)
-            sys.stderr.write(message % self.pidfile)
-            sys.exit(1)
+            # Check if the pid in the pidfile corresponds to a running process
```
Comment (remh) [inline daemon.py]: let's use psutil.pid_exists, it's moe concise, easier to read and works on Windows (even though this part of the code is not run on Windows) http://pythonhosted.org/psutil/#psutil.pid_exists

**Comment by jkoppe** [comment]
Now that datadog-agent runs under supervisord, is it necessary to write pid files like this?

**Comment by alq666** [comment]
@remh is this for CentOS5 that we still carry pid files?

**Comment by remh** [comment]
This pull request is to fix a bug that was introduced with 5.2. 

But you are right a lot of this logic is not needed anymore, we have a card in our board to clean this but it won't be done before 5.4.0.

**Comment by degemer** [comment]
Pidfiles for processes spawned by supervisor were introduced to fix #941 (so that old commands `dd-agent`, `dogstatsd` and `dd-forwarder` still display a correct `status`).

**Comment by jkoppe** [comment]
Gotcha.  This looks like an adequate, simple improvement until the day comes when this code can be ripped out.  Thanks!

**Pull request merged**